tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7693,6 +7693,8 @@ def warn_hlsl_attribute_expects_uint_literal : Warning<
   InGroup<HLSLAttributeType>;
 def err_hlsl_attribute_expects_float_literal : Error<
   "attribute %0 must have a float literal argument">;
+def err_hlsl_wave_matrix_array : Error<
+  "Array of '%0' is not supported">;
```

Comment:
I feel like we have a similar diagnostic for record types on work graph entries. It would be better to combine them and have “%select{}” text to choose different formulations of why arrays are disallowed.

Reply:
Merged with err_hlsl_array_entry_param_disallowed.